Automattic/newspack-plugin
# feat(webhooks): endpoint management ui

### All Submissions:






### Changes proposed in this Pull Request:

Adds UI for managing webhook endpoints:

<img width="708" alt="image" src="https://user-images.githubusercontent.com/820752/208704735-c33a792f-6553-40a0-8ebc-79db0c521e14.png">


<img width="1150" alt="image" src="https://user-images.githubusercontent.com/820752/208166094-4937f972-a289-42e0-8a21-0522ec7dd7a0.png">

<img width="729" alt="Captura de Tela 2022-12-16 às 15 37 53" src="https://user-images.githubusercontent.com/820752/208166637-83511156-e4fa-4614-a7d7-fb0aabbdbadc.png">

<img width="734" alt="Captura de Tela 2022-12-16 às 15 38 34" src="https://user-images.githubusercontent.com/820752/208166657-3ffcbc1c-3c55-4feb-af79-7cfd8eae98e1.png">

<img width="714" alt="image" src="https://user-images.githubusercontent.com/820752/208701700-f8a9bfdd-96f4-45ac-8d46-d1b693c442c9.png">\

<img width="729" alt="image" src="https://user-images.githubusercontent.com/820752/208701626-4b8b128a-7577-458e-8e40-c84ef14692e9.png">

### How to test the changes in this Pull Request:

1. Check out this branch, visit the "Connections" wizard and confirm nothing is changed
2. Add the experimental flag to your config: `define( 'NEWSPACK_EXPERIMENTAL_WEBHOOKS', true );`
3. Refresh the wizard and confirm you see the Webhook Endpoints section at the end
4. Confirm you have AMP Plus, RAS, and Newspack logging enabled:
```php
define( 'NEWSPACK_AMP_PLUS_ENABLED', true );
define( 'NEWSPACK_EXPERIMENTAL_READER_ACTIVATION', true );
define( 'NEWSPACK_LOG_LEVEL', 2 );
```
5. Create your first global webhook endpoint pointing to an URL provided by https://webhook.site/ (visit the site to get one)
6. Create a second webhook endpoint pointing to a non-existent URL
7. In incognito, register a new reader through a RAS flow
8. Refresh the wizard page, click the `info` icon and confirm the webhook requests are scheduled
9. Wait until they are sent, confirm you receive the request and the wizard table is also updated
10. Create a new endpoint and test different test sends before saving. Confirm you receive appropriate error messages.
11. Edit the existing endpoint to a different URL/actions and confirm it's saved
12. Delete the endpoint and refresh to confirm the endpoint is deleted

### Other information:
